### PR TITLE
Added bookmark section in not-logged-in version

### LINKED ISSUE #4250: Bookmarks  when user not logged in.
**Summary:** 
U can still bookmark a picture when ur not logged in from the media details.

**Steps to reproduce:** 
Tap on a media in explore or in uploaded from mobile.
Tap on the star to mark it as bookmark.
The app says "Added to the bookmark".

What did you expect the app to do, and what did you see instead?
User should not be able to bookmark the items 
or
We must include a bookmark section for not logged in User.

**Device and Android version:** 

Nokia 6.1+ android 10.
 
**Commons app version:** 

latest master 3.0.0

**Would you like to work on the issue?**
Yes.

### PR DESCRIPTION
**Pull request by 4D17Y4**
**Description (required)**

Fixes #4250 

What changes did you make and why?

BookmarkPagerFragment 
* Added a constructor parameter which tells Adapter if onlyPicture fragment of bookmark is to initialized (i.e. when user not logged in).
* If onlyPicture is true we do not add location bookmark fragment in the pages.

BookmarkFragment
* Used preferences to check if the user is logged in.
* Added a method to set up the Tab Layout that considers the case when the adapter has only one element.

NavTabLoggedOut
* Added 'favorites' navTabElement.

BookmarksPagerAdaterTests.kt
* Updated the function according to the new constructor.

MainActivity
* Minor inhancement as it showed uploaded from mobile instead of explore.
 

**Tests performed (required)**

Tested betaDebug on Nokia 6.1+ with API level 29.

### CODE REVIEW

**Comment by 4D17Y4** [comment]
Up for review and suggestions.
@neslihanturan, @nicolas-raoul.

File: app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkFragment.java [hunk 1]
```diff
@@ -65,12 +74,26 @@ public View onCreateView(@NonNull final LayoutInflater inflater,
...
     return view;
   }
 
+  /**
+   * This method sets up the tab layout.
+   * If the adapter has only one element it sets the visibility of tabLayout to gone.
+   */
+  public void setupTabLayout(){
+    tabLayout.setVisibility(View.VISIBLE);
+    if(adapter.getCount()==1){
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkFragment.java:91]: We usually put spaces before and after `==` and other logical operators, see for instance the code snippets found at http://www.literateprogramming.com/javaconv.pdf

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Minor style change.
Also, would it make sense to make a new unit test?

**Comment by 4D17Y4** [comment]
Yus @nicolas-raoul ,
I think I can add that.

**Comment by 4D17Y4** [comment]
Added new unit test.

**Review by nicolas-raoul — APPROVED** [review]
Works well, thanks!

**Pull request merged**